packages/language/src/preprocessor/instructions.ts

```diff
@@ -0,0 +1,250 @@
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+import { Token } from "../parser/tokens";
+import * as ast from "../syntax-tree/ast";
+
+export interface InstructionNode {
+  labels: string[];
+  instruction: Instruction;
+  next?: InstructionNode;
+}
+
+export function getLastInstruction(node: InstructionNode): InstructionNode {
```

Comment:
We should probably create an additional data structure that manages the linked list, and makes it easy to retrieve the last element without having to walk it.

Reply:
I've added a `LinkedInstructionList` that keeps track of `head` and `last`.